### PR TITLE
Node A11y Test for Doc Site

### PR DESCRIPTION
**Pull request by AlexBrizi**
### Summary
Added a node script that tests each url on the terra-ui doc site for WCAG2AA and 508 accessibility rules and logs the errors.

Resolves [#138](https://github.com/cerner/terra-ui/issues/138)

### CODE REVIEW

File: scripts/a11y/a11y-output-txt.txt [hunk 1]
```diff
@@ -0,0 +1,2619 @@
+[ { violations:
```
Comment (emilyrohrbough) [inline scripts/a11y/a11y-output-txt.txt]: I don't think we want to check these in

Comment (emilyrohrbough) [reply scripts/a11y/a11y-output-txt.txt]: Also, does puppeter off a way to display this output in a way that is easier to read?

File: scripts/a11y/index.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  browser.close();
+  //axeReports.processResults(promises, 'csv', '/scripts/a11y/test-results', true);
+
+  // write output to file scripts/a11y/a11y-output.txt
+  fs.writeFile("scripts/a11y/a11y-output-txt.txt", util.inspect(results, false, null, true), function(err) {
+    if(err) {
+        return console.log(err);
+    }
+    console.log();
+    console.log("Logs successfully outputted to: terra-core/scripts/a11y");
```
Comment (emilyrohrbough) [inline scripts/a11y/index.js]: We can simplify this to 
```diff
if(err) {
    return console.log(err);
- }
+ } else {
+    console.log("\nLogs successfully outputted to: terra-core/scripts/a11y");
+ }
-    console.log();
-    console.log("Logs successfully outputted to: terra-core/scripts/a11y");
```

File: scripts/a11y/index.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  });
+  const diff = process.hrtime(time);
+  console.log();
+  console.log(`Script run time: ${diff[0]} seconds.`);
```
Comment (emilyrohrbough) [inline scripts/a11y/index.js]: We an use `\n` to add a new line to remove the need to add two `console.log()`s

```diff
- console.log();
- console.log(`Script run time: ${diff[0]} seconds.`);
+ console.log(`\nScript run time: ${diff[0]} seconds.`);
```

File: scripts/a11y/index.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      // add axe-core to the pages
+      await page.addScriptTag({
+        path: require.resolve('axe-core')
+      });
+      console.log('[' + i + '/' + urls.length-1 + ']   ', url);
+      // run axe on the page
+      axeResults = await page.evaluate(async () => {
+        return await axe.run(document, {
+          runOnly: ['wcag2aa', 'section508'],
+          resultTypes: ['violations']
```
Comment (emilyrohrbough) [inline scripts/a11y/index.js]: do we want to limit the reports to violations?

File: scripts/a11y/index.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        });
+      });
+      delete axeResults.incomplete;
```
Comment (emilyrohrbough) [inline scripts/a11y/index.js]: We should probably track anything that returned incompelte

File: package.json [hunk 6]
```diff
@@ -37,8 +38,12 @@
     "deploy": "gh-pages -d build"
   },
   "dependencies": {
+    "axe-core": "^3.1.2",
+    "axe-reports": "^1.1.11",
     "classnames": "^2.2.5",
+    "fs": "0.0.1-security",
```
Comment (emilyrohrbough) [inline package.json]: Do we need to define this dependency?

Comment (bjankord) [reply package.json]: No, `fs` comes as a built with node.

File: scripts/a11y/a11y-output-txt.txt (hunk 1 above)
Comment (AlexBrizi) [reply scripts/a11y/a11y-output-txt.txt]: I made it so the output is printed as both a .txt file and .csv file and included it in the PR (see files changed).

File: scripts/a11y/index.js (hunk 4 above)
Comment (bjankord) [reply scripts/a11y/index.js]: Yeah, I believe so. I don't think the other reports are valuable in this context.

**Pull request merged**